src/MCPServer/tests/test_package.py

```diff
@@ -40,3 +51,80 @@ def test__determine_transfer_paths(name, path, tmpdir, expected):
     assert results[0] == expected[0], "name mismatch"
     assert results[1] == expected[1], "path mismatch"
     assert results[2] == expected[2], "tmpdir mismatch"
+
+
+class TestPadDestinationFilePath:
+    def test_zipfile_does_not_exist(self, tmp_path):
+        transfer_path = tmp_path / "transfer.zip"
+        padded_path = _pad_destination_filepath_if_it_already_exists(str(transfer_path))
+        assert padded_path == str(transfer_path)
+
+    def test_zipfile_is_padded_if_exists(self, tmp_path):
+        transfer_path = tmp_path / "transfer.zip"
+        transfer_path.touch()
+        padded_path = _pad_destination_filepath_if_it_already_exists(str(transfer_path))
+        assert padded_path == str(tmp_path / "transfer_1.zip")
+
+    def test_dir_does_not_exist(self, tmp_path):
+        transfer_path = tmp_path / "transfer/"
+        padded_path = _pad_destination_filepath_if_it_already_exists(str(transfer_path))
+        assert padded_path == str(transfer_path)
+
+    def test_dir_is_padded_if_exists(self, tmp_path):
+        transfer_path = tmp_path / "transfer/"
+        transfer_path.mkdir()
+        padded_path = _pad_destination_filepath_if_it_already_exists(str(transfer_path))
+        assert padded_path == str(tmp_path / "transfer_1")
```

Comment:
Minor: I find it odd that you're testing an `_` prefix function; that feels like you're testing an implementation detail. Either make this a public function of a utility package, or treat it as an internal function that doesn't get tested directly.

(LMK if that's not clear, I am struggling at words today.)

Reply:
Hmm, I see what you're saying. It does feel like it's a unit that should be tested, and I'd rather test all the possibilities at this level than work them into more complex tests of the calling functions. I'm inclined to go with the second option, but it'd be good to know what the Artefactual devs think :-)

Reply:
We don't have a hard rule there. We're still at that point where I believe that any test is better than no test - so I'm personally pretty open about it. Also I trust whatever is your preference, I'm mostly learning here. I haven't had many chances before to see the value of different approaches compared. I've definitely written tests like yours before before, also when trying to understand a function that I have no idea what is doing.